### DISCUSSION TITLE
IP ranges

### DISCUSSION THREAD

**Discussion by candlerb** [netbox-community/netbox#7560] [Ideas]
I have a few minor points relating to IP ranges, which I thought I'd bring up as discussions rather than issues for now.

(1) I'd like to be able to record the usage of a role, particularly as "DHCP pool".  Other applications for ranges include a pools of addresses for auto-assignment (e.g. [metallb](https://metallb.universe.tf/configuration/#layer-2-configuration) can be given a pool with start and end address), or pools for manual assignment (e.g. switches, printers etc)

Currently the "Roles" for IP Ranges are "Backup, Development, Management, Production" - which match with Prefixes - and status "Active", "Reserved", "Deprecated"

I guess I can use tags.  But I wonder if it might be possible to define more useful Roles for ranges which are still sufficiently generic, e.g. "DHCP", "Auto assign", "Manual assign", "Loopbacks"

(2) Ranges are not visible within IP Address listings.

For example, suppose I create a range 10.12.255.240 - 10.12.255.249.  When I view the IP addresses under prefix 10.12.255.0/24, I see:

![image](https://user-images.githubusercontent.com/44789/137696522-cbc8783d-f9a7-4ee1-9ae9-648146d4e2f1.png)

There's no indication that the addresses .240-.249 are already assigned to a range. Indeed, the range is already counted towards prefix utilisation, so addresses .240-.249 are not "available" by that definition; but the link says that 153 addresses from .100 to .252 are available.

(3) When viewing a range, it would be nice to be able to see and navigate to "parent prefixes", as you can with IP addresses.  That would make it much easier to get to the IP addresses in the same subnet.

Any thoughts?

**Comment by kkthxbye-code** [comment]
>Currently the "Roles" for IP Ranges are "Backup, Development, Management, Production" - which match with Prefixes

These roles are user-defined, you can add/edit/delete under /ipam/roles/.

**Reply by candlerb** [reply]
I see it now - "Prefix & VLAN Roles" it says on the side bar. Thanks!

**Comment by laherre** [comment]
What about the question (2) , and also the child prefixes are not shown in then IP Address listings

> (2) Ranges are not visible within IP Address listings.
> 
> For example, suppose I create a range 10.12.255.240 - 10.12.255.249. When I view the IP addresses under prefix 10.12.255.0/24, I see:
> 
> There's no indication that the addresses .240-.249 are already assigned to a range. Indeed, the range is already counted towards prefix utilisation, so addresses .240-.249 are not "available" by that definition; but the link says that 153 addresses from .100 to .252 are available.

**Reply by koratfood** [reply]
> Ranges are not visible within IP Address listings.

If you tick the [Mark Populated](https://netboxlabs.com/docs/netbox/models/ipam/iprange/#mark-populated) option, you should get the desired view.

> the child prefixes are not shown in then IP Address listings

As far as I know, Child Prefixes were never meant to be visible under the IP Addresses tab, since they are *not* IP (host) Addresses.
